src/xtgeo/xyz/_xyz.py

```diff
@@ -103,15 +103,30 @@ def _df_column_rename(self, newname, oldname):
         else:
             raise ValueError(f"Wrong type of input to {newname}; must be string")
 
-    def _check_name(self, value):
-        if not isinstance(value, str):
-            raise ValueError(f"Wrong type of input; must be string, was {type(value)}")
+    def _check_name_and_replace(self, oldname, newname):
+        """Replace name of a column, doing some checks."""
+        if not isinstance(newname, str):
+            raise ValueError(
+                f"Wrong type of input; must be string, was {type(newname)}"
+            )
 
-        if value not in self.get_dataframe(copy=False).columns:
+        if oldname not in self.get_dataframe(copy=False).columns:
             raise ValueError(
-                f"{value} does not exist as a column name, must be "
-                f"one of: f{self.get_dataframe(copy=False).columns}"
+                f"The column {oldname} does not exist as a column name, must be "
+                f"one of: f{self.get_dataframe(copy=False).columns}. Likely a bug!"
             )
+        # check that newname not already there; will be ambigious
+        if newname in self.get_dataframe(copy=False).columns:
+            raise ValueError(
+                f"The column {newname} does already exist as a column name:"
+                f"f{self.get_dataframe(copy=False).columns}. The new name "
+                "must be unique!"
+            )
+
+        # replace column name
+        df = self.get_dataframe(copy=True)
+        df.rename(columns={oldname: newname}, inplace=True)
```

Comment:
We should probably avoid the use of the `inplace` argument, they have been talking about deprecating it for some time https://github.com/pandas-dev/pandas/issues/16529
```suggestion
        df = df.rename(columns={oldname: newname})
```

Reply:
Thanks, you are right but there are numerous places in xtgeo then which could be updated on this topic, and probably best to that in a separate PR. An alternative I just made an issue for now, is to shift to Polars in stead. See issue https://github.com/equinor/xtgeo/issues/1314